File: scripts/copy-pngs.py
import shutil, os, json

src = r"C:\Users\mikej\OneDrive\Documents\apps\pedal-sheet\public"
dst = "vtt/public/images"

portrait_names = ["kehrfuffle", "strider", "toern", "wendy"]

categories = {"battlemaps": [], "portraits": [], "tokens": [], "items": []}

os.makedirs(f"{dst}/battlemaps", exist_ok=True)
os.makedirs(f"{dst}/portraits", exist_ok=True)
os.makedirs(f"{dst}/tokens", exist_ok=True)
os.makedirs(f"{dst}/items", exist_ok=True)

count = 0
for fname in os.listdir(src):
    if not fname.endswith(".png") or fname in ("icon.png", "t_pin.png"):
        continue
    base = fname.replace(".png", "")
    if fname.endswith("_enc.png"):
        folder = "battlemaps"
    elif fname.endswith("_bm.png"):
        folder = "tokens"
    elif base.lower() in portrait_names:
        folder = "portraits"
    else:
        folder = "items"
    shutil.copy2(os.path.join(src, fname), os.path.join(dst, folder, fname))
    categories[folder].append(f"/images/{folder}/{fname}")
    count += 1
    print(f"OK: {fname} -> {folder}/")

# Update manifest
with open(f"{dst}/../image-manifest.json") as f:
    manifest = json.load(f)

for cat, files in categories.items():
    if files:
        manifest[cat] = list(set(files + manifest.get(cat, [])))

with open(f"{dst}/../image-manifest.json", "w") as f:
    json.dump(manifest, f, indent=2)

print(f"\nTotal: {count} files copied")
for cat, files in categories.items():
    if files:
        print(f"  {cat}: {len(files)}")

File: scripts/copy_pngs.mjs
/**
 * Copies PNGs from the pedal-sheet project to the VTT project.
 * Naming convention:
 *   *_enc.png → battlemaps/
 *   *_bm.png  → tokens/
 *   PC names (kehrfuffle, strider, toern, wendy) → portraits/
 *   Rest → items/ (for homebrew/inventory)
 */
import fs from 'fs';
import path from 'path';

const SRC = 'C:/Users/mikej/OneDrive/Documents/apps/pedal-sheet/public';
const DST = path.resolve('vtt/public/images');

const categories = {
  battlemaps: [],
  portraits: [],
  tokens: [],
  items: [],
};

const portraitNames = ['kehrfuffle', 'strider', 'toern', 'wendy'];

fs.readdirSync(SRC).forEach(file => {
  if (!file.endsWith('.png')) return;

  let destFolder;
  const base = file.replace('.png', '');

  if (file.endsWith('_enc.png')) {
    destFolder = 'battlemaps';
  } else if (file.endsWith('_bm.png')) {
    destFolder = 'tokens';
  } else if (portraitNames.includes(base.toLowerCase())) {
    destFolder = 'portraits';
  } else {
    destFolder = 'items';
  }

  const srcPath = path.join(SRC, file);
  const dstPath = path.join(DST, destFolder, file);

  fs.cpSync(srcPath, dstPath, { force: true });
  categories[destFolder].push(`/images/${destFolder}/${file}`);
  console.log(`✅ Copied: ${file} → images/${destFolder}/`);
});

// Read existing manifest
const manifestPath = path.join(DST, '..', 'image-manifest.json');
const existing = JSON.parse(fs.readFileSync(manifestPath, 'utf8'));

// Merge new files into manifest
for (const [cat, files] of Object.entries(categories)) {
  if (files.length > 0) {
    existing[cat] = [...new Set([...files, ...(existing[cat] || [])])];
  }
}

fs.writeFileSync(manifestPath, JSON.stringify(existing, null, 2));
console.log('\n📝 Updated image-manifest.json');

// Summary
console.log('\n=== COPY SUMMARY ===');
for (const [cat, files] of Object.entries(categories)) {
  if (files.length > 0) {
    console.log(`  ${cat}: ${files.length} files`);
  }
}
console.log('\nDone!');
